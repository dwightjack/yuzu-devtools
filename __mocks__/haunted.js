const haunted = require('haunted/index');

module.exports = {
  ...haunted,
  useContext: jest.fn(),
};
