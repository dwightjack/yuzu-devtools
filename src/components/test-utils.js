export const mockComponent = (tag) => (props) => {
  const el = document.createElement(tag);
  Object.keys(props).forEach((key) => {
    el.setAttribute(key, props[key]);
  });
  return el;
};
