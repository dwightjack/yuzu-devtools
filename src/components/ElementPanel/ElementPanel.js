import { html } from 'lit-html';
import { component, useContext } from 'haunted';
import '../SidePanel/SidePanel';
import '../BlankSlate/BlankSlate';
import '../PropList/PropList';
import { StateContext } from '../../store/stateContext';

export default function ElementPanel() {
  const {
    Component: name,
    uid,
    options = {},
    state = {},
    watchers,
    onPropCheck,
  } = useContext(StateContext);

  return html`
    <style>
      .title {
        margin: 0;
        padding: 8px var(--gutter);
        font-size: var(--font-size-l);
        font-weight: bold;
        line-height: 1.2;
        background-color: var(--color-lighter);
      }

      .title > em {
        font-size: 0.75em;
      }
    </style>
    <yzdt-side-panel>
      ${!uid
        ? html`
            <yzdt-blank-slate slot="body"
              >Select a component on the left panel to inspect its
              properties</yzdt-blank-slate
            >
          `
        : html`
            <h2 slot="header" class="title">
              ${name || 'Component'}<em>#${uid}</em>
            </h2>
            <div slot="body">
              <yzdt-prop-list name="Options" .props=${options}></yzdt-prop-list>
              <yzdt-prop-list
                name="State"
                uid=${uid}
                ?watchable=${true}
                .onSelect=${onPropCheck}
                .props=${state}
                .watchers=${watchers}
              ></yzdt-prop-list>
            </div>
          `}
    </yzdt-side-panel>
  `;
}

// ElementPanel.observedAttributes = ['name', 'uid'];
customElements.define('yzdt-element-panel', component(ElementPanel));
