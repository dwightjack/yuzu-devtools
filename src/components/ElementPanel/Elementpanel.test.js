import * as haunted from 'haunted';
import { toHTML } from '../test-utils';
import ElementPanel from './ElementPanel';

jest.mock('haunted');

const data = { uid: '_0', Component: 'DEMO' };

describe('ElementPanel', () => {
  beforeEach(() => {
    haunted.useContext.mockImplementation(() => data);
  });
  test('default snapshot', () => {
    const { html } = toHTML(ElementPanel());
    expect(html).toMatchSnapshot();
  });
  test('empty snapshot', () => {
    haunted.useContext.mockImplementationOnce(() => ({}));
    const { html } = toHTML(ElementPanel());
    expect(html).toMatchSnapshot();
  });
});
