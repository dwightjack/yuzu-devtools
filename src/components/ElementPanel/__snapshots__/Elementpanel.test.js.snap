// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ElementPanel default snapshot 1`] = `
<yzdt-side-panel>
  <h2 slot="header"
      class="title"
  >
    DEMO
    <em>
      #_0
    </em>
  </h2>
  <slot slot="body">
  </slot>
</yzdt-side-panel>
`;

exports[`ElementPanel empty snapshot 1`] = `
<yzdt-side-panel>
  <yzdt-blank-slate slot="body">
    Select a component on the left panel to inspect its
              properties
  </yzdt-blank-slate>
</yzdt-side-panel>
`;
